source/MaterialXGenShader/TypeDesc.cpp

```diff
@@ -35,12 +36,25 @@ const TypeDesc* TypeDesc::registerType(const string& name, unsigned char basetyp
         throw ExceptionShaderGenError("A type with name '" + name + "' is already registered");
     }
 
-    TypeDescPtr ptr = TypeDescPtr(new TypeDesc(name, basetype, semantic, size, editable));
+    TypeDescPtr ptr = TypeDescPtr(new TypeDesc(name, basetype, semantic, size, editable, channelMapping));
     map[name] = ptr;
 
     return ptr.get();
 }
 
+int TypeDesc::getChannelIndex(char channel) const
+{
+    auto it =_channelMapping.find(channel);
+    if (it != _channelMapping.end())
+    {
+        return it->second;
+    }
+    else
+    {
+        return -1;
+    }
```

Comment:
Nitpick: the if-else could be a one line: return it != _channelMapping.end() ? it->second : -1;